addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java

```diff
@@ -472,6 +472,9 @@ private void updateScreenSaverStatus(boolean screenSaverActive) {
     }
 
     public synchronized void close() {
+        if (socket.isConnected()) {
```

Comment:
maybe a null check here? I don't know whether its guaranteed that close is only called once?

Reply:
Hm, ... Yes, that would be much more safer.
But maybe we should avoid setting the socket to `null` anyway, because we have a lot more lines in the code to check to make it really safe.